consensus/ibft/extra_test.go

```diff
@@ -40,3 +58,135 @@ func TestExtraEncoding(t *testing.T) {
 		}
 	}
 }
+
+func generateKeysAndAddresses(t *testing.T, num int) ([]*ecdsa.PrivateKey, []types.Address) {
+	t.Helper()
+
+	keys := make([]*ecdsa.PrivateKey, num)
+	addrs := make([]types.Address, num)
+
+	for i := range keys {
+		pk, addr := tests.GenerateKeyAndAddr(t)
+		keys[i] = pk
+		addrs[i] = addr
+	}
+
+	return keys, addrs
+}
+
+func createIBFTHeader(
+	t *testing.T,
+	num uint64,
+	parentHash types.Hash,
+	validators []types.Address,
+	lastCommittedSeal [][]byte,
+) *types.Header {
+	t.Helper()
+
+	header := &types.Header{
+		Number:     num,
+		ParentHash: parentHash,
+	}
+
+	assert.NoError(t, initIbftExtra(header, validators, lastCommittedSeal))
+
+	return header
+}
+
+// Test Scenario
+// 1. 4 IBFT Validators create headers
+// 2. A faulty node scans the past headers and appends new committed seal
+// 3. Check each hashes of headers are wrong
+func TestAppendCommittedSeal(t *testing.T) {
+	useIstanbulHeaderHash(t)
+
+	var (
+		numHeaders          = 5
+		numNormalValidators = 4
+		numFaultyValidators = 1
+
+		headers                      = make([]*types.Header, 0, numHeaders)
+		faultyHeaders                = make([]*types.Header, 0, numHeaders)
+		parentHash                   = types.StringToHash("genesis")
+		parentCommittedSeal [][]byte = nil
+
+		keys, addresses     = generateKeysAndAddresses(t, numNormalValidators+numFaultyValidators)
+		normalValidatorKeys = keys[:numNormalValidators]
+		faultyValidatorKey  = keys[numNormalValidators]
+
+		err error
+	)
+
+	// create headers by normal validators
+	for i := 0; i < numHeaders; i++ {
+		header := createIBFTHeader(t, uint64(i+1), parentHash, addresses, parentCommittedSeal)
+
+		// write seal
+		header, err = writeSeal(keys[0], header)
+		assert.NoError(t, err)
+
+		// write committed seal
+		committedSeal := make([][]byte, len(normalValidatorKeys))
+		for i, key := range normalValidatorKeys {
+			committedSeal[i], err = createCommittedSeal(key, header)
+			assert.NoError(t, err)
+		}
+
+		assert.NoError(t, packCommittedSealIntoIbftExtra(header, committedSeal))
+
+		header = header.ComputeHash()
+
+		headers = append(headers, header)
+
+		parentHash = header.Hash
+		parentCommittedSeal = committedSeal
+	}
+
+	// faulty node scans the past headers and try to inject new committed seal
+	for i, h := range headers {
+		header := h.Copy()
+
+		// update parent hash & committed seal
+		if i > 0 {
+			parentHeader := faultyHeaders[i-1]
+
+			// update parent hash
+			header.ParentHash = parentHeader.Hash
+
+			// get parent committeed seal
```

Comment:
Typo here, should be `committed`

Reply:
Thank you! Fixed at 9192f447dc5a4e63cbceb1500a1690b6283d385d